theano/compile/mode.py

```diff
@@ -199,7 +199,7 @@ def apply(self, fgraph):
 
 # rearranges elemwise expressions
 optdb.register('canonicalize', gof.EquilibriumDB(ignore_newtrees=False),
-               1, 'fast_run', 'fast_compile')
+               1, 'fast_run', 'fast_compile', 'canonicalize_db')
```

Comment:
Good idea, but I would have named it canonicalize_eq. @lamblin what do you think? I can change this in another PR.